src/jws/verify.rs

```diff
@@ -0,0 +1,544 @@
+use askar_crypto::sign::KeySigVerify;
+
+use crate::{
+    error::{err_msg, ErrorKind, Result, ResultExt},
+    jws::ParsedJWS,
+};
+
+impl<'a, 'b> ParsedJWS<'a, 'b> {
+    pub(crate) fn verify<Key: KeySigVerify>(&self, signer: (&str, &Key)) -> Result<bool> {
+        let (kid, key) = signer;
+
+        let (i, signature) = self
+            .jws
+            .signatures
+            .iter()
+            .enumerate()
+            .find(|(_, sig)| sig.header.kid == kid)
+            .ok_or_else(|| err_msg(ErrorKind::InvalidState, "KID not found"))?;
+
+        let protected = self
```

Comment:
I know that we support only one signature, but  I don't like the idea that the order of the signatures must match the order of the header.